lib/core/src/Cardano/Wallet.hs

```diff
@@ -811,49 +870,118 @@ signTx
+
             let keyFrom = isOwned (getState cp) (xprv, pwd)
-            (tx, wit) <- either
-                (throwE . ErrSignTx)
-                pure
-                (mkStdTx tl keyFrom ins allShuffledOuts )
-            withExceptT ErrSignTxNoSuchWallet $
+            (tx, sealedTx) <- withExceptT ErrSignPaymentMkTx $ ExceptT $ pure $
+                mkStdTx tl keyFrom ins allOuts
+
+            let bp = blockchainParameters cp
+            let (time, meta) = mkTxMeta bp (currentTip cp) s' ins allOuts
+            return (tx, meta, time, sealedTx)
+  where
+    db = ctx ^. dbLayer @s @k
+    tl = ctx ^. transactionLayer @t @k
+
+signDelegation
+    :: forall ctx s t k.
+        ( HasTransactionLayer t k ctx
+        , HasDBLayer s k ctx
+        , Show s
+        , NFData s
+        , IsOwned s k
+        , GenChange s
+        , HardDerivation k
+        , AddressIndexDerivationType k ~ 'Soft
+        , WalletKey k
+        )
+    => ctx
+    -> WalletId
+    -> ArgGenChange s
+    -> Passphrase "encryption"
+    -> CoinSelection
+    -> PoolId
+    -> ExceptT ErrSignDelegation IO (Tx, TxMeta, UTCTime, SealedTx)
+signDelegation ctx wid argGenChange pwd coinSel poolId = db & \DBLayer{..} -> do
+    let (CoinSelection ins outs chgs) = coinSel
+    withRootKey @_ @s ctx wid pwd ErrSignDelegationWithRootKey $ \xprv -> do
+        mapExceptT atomically $ do
+            cp <- withExceptT ErrSignDelegationNoSuchWallet $ withNoSuchWallet wid $
+                readCheckpoint (PrimaryKey wid)
+            (allOuts, s') <-
+                assignChangeAddresses argGenChange outs chgs (getState cp)
+            withExceptT ErrSignDelegationNoSuchWallet $
                 putCheckpoint (PrimaryKey wid) (updateState s' cp)
-            let ourCoins (TxOut addr (Coin val)) =
-                    if fst (isOurs addr s')
-                        then Just (fromIntegral val)
-                        else Nothing
-            let amtOuts =
-                    sum (mapMaybe ourCoins allShuffledOuts)
-            let amtInps = fromIntegral $
-                    sum (getCoin . coin . snd <$> ins)
-            let txSlot =
-                    (currentTip cp) ^. #slotId
-            let meta = TxMeta
-                    { status = Pending
-                    , direction = Outgoing
-                    , slotId = txSlot
-                    , blockHeight = (currentTip cp) ^. #blockHeight
-                    , amount = Quantity (amtInps - amtOuts)
-                    }
-            let time = slotStartTime
-                    (slotParams (blockchainParameters cp))
-                    txSlot
-            return (tx, meta, time, wit)
+
+            -- TODO
+            -- Pre-compute this and store it with the wallet static metadata.
+            let rewardAccount = deriveRewardAccount @k xprv pwd
+
+            let hrp = Bech32.unsafeHumanReadablePartFromText "addr"
+            let accountAddr
```

Comment:
Is importing `Bech32` and knowing how to create the reward `Address` in `core` is temporary? (it is target-specific)

Reply:
![Screenshot from 2019-11-27 19-17-39](https://user-images.githubusercontent.com/5680256/69749358-a1525100-114a-11ea-8d1d-c040edad11a6.png)

It is, as indicated by this big fat "TEMPORARY" in the corresponding commit :sweat_smile: